GoogleIoTCore.agent.nut

```
@@ -0,0 +1,1004 @@
+        _onDisconnectedCb = callback;
+    }
+
+    function setDebug(value) {
+        _debugEnabled = value;
+        _transport && _transport._setDebug(_debugEnabled);
+    }
+    // -------------------- PRIVATE METHODS -------------------- //
+
+    function _onConnected(err) {
+        _onConnectedCb && _onConnectedCb(err);
+    }
+
+    function _onDisconnected(reason) {
+        _onDisconnectedCb && _onDisconnectedCb(reason);
+    }
+
+    function _getOAuthToken(iss, secret, callback) {
+        local config = {
+            "iss"         : iss,
+            "jwtSignKey"  : secret,
+            "scope"       : "https://www.googleapis.com/auth/cloud-platform"
+        };
+        // Initializing client with provided Google Firebase config
+        local client = OAuth2.JWTProfile.Client(OAuth2.DeviceFlow.GOOGLE, config);
+
+        // TODO: Fix it in the OAuth lib and remove this line
+        client._debug = false;
+
+        // Starting procedure of access token acquisition
+        client.acquireAccessToken(
+            function(resp, err) {
+                if (err) {
+                    _logError("OAuth token acquisition error: " + err);
+                    callback(GOOGLE_IOT_CORE_ERROR_GENERAL, null);
+                } else {
+                    callback(0, resp);
+                }
+            }.bindenv(this)
+        );
+    }
+
+    function _getDevice(oatoken, callback) {
+        local headers = {
+            "authorization" : "Bearer " + oatoken,
+            "content-type" : "application/json",
+            "cache-control" : "no-cache"
+        };
+
+        local url = "https://cloudiot.googleapis.com/v1/";
+        url += format("projects/%s/locations/%s/registries/%s/devices/%s", _projectId, _cloudRegion, _registryId, _deviceId);
+
+        local req = http.get(url, headers);
+
+        local sent = function (resp) {
+            if (resp.statuscode == 404) {
+                callback(0, null);
```

Don't we want to return an error in this case?

---

Why do we make this exception for 404?